freqtrade/optimize/backtesting.py

```diff
@@ -960,13 +998,16 @@ def backtest(self, processed: Dict,
 
                 for t in list(open_trades[pair]):
                     # 1. Cancel expired entry/exit orders.
-                    if self.check_order_cancel(t, current_time):
-                        # Close trade due to entry timeout expiration.
+                    order_cancel = self.check_order_cancel(t, current_time)
+                    # 2. Replace/cancel (user requested) entry orders.
+                    order_replace = self.check_order_replace(t, current_time, row)
+                    if order_cancel or order_replace:
```

Comment:
Behavior in live (freqtradebot.py) and backtesting (here) will be slightly different.

Live will only call the new callback if the order was NOT cancelled. Here, you always call both callbacks. 

I also think that this condition can lead to odd circumstances 
e.g. check_cancel cancels the order - and order_replace replaces it.
As `order_cancel` will then still be True - the trade will be deleted - so the whole "replace" logic was for nothing.


You can probably get arount this by combining the two into the `if` statement - if the first evaluates to true, the 2nd will not be executed (eager execution)

Reply:
Yeah. I've picked up on it also but thougth it to be harmless. My thinking was this:
1. if `check_order_cancel()` deletes the order from the trade it will no longer execute the callback from `check_order_replace()` since the loop will not find any open orders
2. if `check_order_cancel()` decides to delete the trade alltogether then it doesn't really matter what `check_order_replace()` does (and it will not do much because of 1.)
3. my main priority here was to have timeout check execute before replacement logic

Reply:
ah right - that might work - but i always find such implicit logic (the 2nd won't do anything because there's no order left) dangerous ...

Reply:
Addressed this through https://github.com/freqtrade/freqtrade/pull/6692/commits/4e43194dfeefb135e3a567659dba605bb501b10d.